### ISSUE TITLE
Infinite loop when using Buf or Blob as a NativeCall Return Type

### ISSUE THREAD

**Issue by Xliff** [rakudo/rakudo#3630]
Given the following piece of code, rakudo will never return:

perl6 -e 'use NativeCall; sub malloc(uint64) is native returns Buf { * }; my $a = malloc(100); say $a.^name;'

However this code works:

perl6 -e 'use NativeCall; sub malloc(uint32) is native returns Pointer { * }; my $a = malloc(100); say $a.^name;'

[This discussion](https://colabti.org/irclogger/irclogger_log/raku-dev?date=2020-04-17#l105) in IRC has the relevant details:

Simply put: Blob and Buf are not valid return types for NativeCall

**Comment by dogbert17** [comment]
The first example now looks like this:

    dogbert@dogbert-VirtualBox:~$ perl6 -v
    Welcome to 𝐑𝐚𝐤𝐮𝐝𝐨™ v2020.09-92-g6e6c9e5cc.
    Implementing the 𝐑𝐚𝐤𝐮™ programming language v6.d.
    Built on MoarVM version 2020.09-20-gf99bc2ce7.
    dogbert@dogbert-VirtualBox:~$ perl6 -e 'use NativeCall; sub malloc(uint64) is native returns Buf { * }; my $a = malloc(100); say $a.^name;'
    Internal error: unhandled dyncall return type
      in block malloc at /home/dogbert/.rakudobrew/versions/moar-master/install/share/perl6/core/sources/947BDAB9F96E0E5FCCB383124F923A6BF6F8D76B (NativeCall) line 592
      in block <unit> at -e line 1
